src/pdb/mod.rs

```diff
@@ -234,31 +235,131 @@ impl PageFlags {
     pub fn page_has_data(&self) -> bool {
         (self.0 & 0x40) == 0
     }
+
+    #[must_use]
+    pub fn is_index_page(&self) -> bool {
+        self.0 == 0x64
+    }
+}
+
+/// An entry in an index page.
+#[binrw]
+#[derive(PartialEq, Eq, Clone, Copy)]
+#[brw(little)]
+pub struct IndexEntry(u32);
+
+impl IndexEntry {
+    /// Returns the page index from bits 31-3.
+    #[must_use]
+    pub fn page_index(&self) -> PageIndex {
+        PageIndex(self.0 >> 3)
+    }
+
+    /// Returns the index flags from bits 2-0.
+    #[must_use]
+    pub fn index_flags(&self) -> u8 {
+        (self.0 & 0b111) as u8
+    }
+
+    /// Returns `true` if the entry is an empty slot.
+    #[must_use]
+    pub fn is_empty(&self) -> bool {
+        self.0 == 0x1FFF_FFF8
+    }
 }
 
-fn write_page_contents<W: Write + Seek>(
-    row_groups: &Vec<RowGroup>,
-    writer: &mut W,
-    endian: Endian,
-    args: (u32,),
-) -> BinResult<()> {
-    let (page_size,) = args;
+impl fmt::Debug for IndexEntry {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if self.is_empty() {
+            f.debug_struct("IndexEntry")
+                .field("is_empty", &self.is_empty())
+                .finish()
+        } else {
+            f.debug_struct("IndexEntry")
+                .field("is_empty", &self.is_empty())
+                .field("page_index", &self.page_index())
+                .field("index_flags", &self.index_flags())
+                .finish()
+        }
+    }
+}
 
-    let header_end_pos = writer.stream_position()?;
+/// The content of an index page.
+#[binrw]
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[brw(little)]
+pub struct IndexPageContent {
+    /// Unknown field, usually `1fff` or `0001`.
+    pub unknown_a: u16,
+    /// Unknown field, usually `1fff` or `0000`.
+    pub unknown_b: u16,
+    /// Magic value, must be `0x03ec`.
+    #[br(temp, assert(magic1 == 0x03ec))]
+    #[bw(calc = 0x03ec)]
+    magic1: u16,
```

Comment:
any reason you chose this formulation instead of just

```suggestion
    #[brw(magic(0x03ec))]
```

Reply:
Well mainly because it is used in the magic value in `pub struct Page` and I was not aware of the existence of that formulation you just suggested. Shall I change that one too?

Reply:
yes please.

Reply:
Done, also replaced in 2 magic fields found in the `Header` struct. Not 100% sure if this is desired, I can amend if necessary